src/core/containers/HandleLogin/index.js

```diff
@@ -58,7 +59,14 @@ function createLoadData(dispatch) {
             secure: config.get('cookieSecure'),
             maxAge: config.get('cookieMaxAge'),
           });
-          router.push('/search');
+          let to;
+          try {
+            to = browserBase64Decode(state.split(':')[1]);
```

Comment:
The FxA state is stored as: `<CSRF-token>:<base64EncodedNextPath>`. The assumption was that if we needed to add more data later we could append another `:` or JSON decode and on error treat it like a next path.

If we were to encode some extra data in here we'd probably want to sign it in which case a cookie might be a better choice for the added space allotment.

We also discussed wanting the CSRF token before the first `:` so that if we run over our allowed space and it gets truncated the login will still work but it might redirect funny or something. So base64 encoding a JSON object isn't quite what we want since that will fail if this gets truncated.